src/agoradatatools/etl/transform/utils.py

```diff
@@ -0,0 +1,197 @@
+
+    discard, obj["bins"] = list(
+        pd.cut(distribution, bins=10, precision=3, retbins=True)
+    )
+    obj["bins"] = np.around(obj["bins"].tolist()[1:], 2)
+    base = [0, *obj["bins"][:-1]]
+    obj["bins"] = zip(base, obj["bins"])
+    obj["bins"] = list(obj["bins"])
+
+    obj["min"] = np.around(df[col].min(), 4)
+    obj["max"] = np.around(df[col].max(), 4)
+    obj["mean"] = np.around(df[col].mean(), 4)
+    obj["first_quartile"] = np.around(
+        df[col].quantile(q=0.25, interpolation="midpoint")
+    )
+    obj["third_quartile"] = np.around(
+        df[col].quantile(q=0.75, interpolation="midpoint")
+    )
+
+    return obj
+
+
+def count_grouped_total(
+    df: pd.DataFrame, grouping: [str, list], input_colname: str, output_colname: str
+) -> pd.DataFrame:
+    """For each unique item/combination in the column(s) specified by grouping,
+    counts the number of unique items in the column [input_colname] that
+    correspond to that grouping. The calculated counts are put in a new
+    column and named with [output_colname].
+    Args:
+        df (pd.DataFrame): contains columns listed in grouping and
+                           input_colname. May contain other columns as well, but
+                           these will be dropped from the returned data frame.
+        grouping (str or list): a string with a single column name, or a list of
+                                strings for multiple column names
+        input_colname (str): the name of the column to count
+        output_colname (str): the name of the new column with calculated counts
+    Returns:
+        pd.DataFrame: a data frame containing the grouping column(s) and a
+                      new column for output_colname, which contains the count of
+                      unique items in input_colname for each grouping item.
+    """
+    df = (
+        df.groupby(grouping)[input_colname]
+        .nunique()
+        .reset_index()
+        .rename(columns={input_colname: output_colname})
+    )
+    return df
+
+
+def transform_overall_scores(df: pd.DataFrame) -> pd.DataFrame:
+    interesting_columns = [
+        "ensg",
+        "hgnc_gene_id",
+        "overall",
+        "geneticsscore",
+        "omicsscore",
+        "literaturescore",
+    ]
+
+    # create mapping to deal with missing values as they take different shape across the fields
+    scored = ["isscored_genetics", "isscored_omics", "isscored_lit"]
+    mapping = dict(zip(interesting_columns[3:], scored))
+
+    for field, is_scored in mapping.items():
+        df.loc[lambda row: row[is_scored] == "N", field] = np.nan
+
+    # LiteratureScore is a string in the source file, so convert to numeric
+    df["literaturescore"] = pd.to_numeric(df["literaturescore"])
+
+    # Remove identical rows (see AG-826)
+    return df[interesting_columns].drop_duplicates()
+
+
+def join_datasets(left: pd.DataFrame, right: pd.DataFrame, how: str, on: str):
```

Comment:
This function has been hanging around in the repo, but it's only used in one place (end of `transform_team_info`). The arguments to this function are identical to the arguments for `pd.merge` so I'm actually not sure why this function exists instead of just directly calling `pd.merge`... can we just delete it and change that one line in `transform_team_info`?

Reply:
Awesome - I think it might help to have a ticket that looks to clean up/refactor the utils.py functions.

Reply:
Agreed!